client/src/js/components/Dashboard/DisplayMessage.js

```diff
@@ -0,0 +1,56 @@
+import React, {Component} from 'react'
+import ReadMessage from './ReadMessage'
+import ShowMessage from './ShowMessage'
+import ArchivedBox from './ArchivedBox'
+import MessageBoard from './MessageBoard'
+
+import AppActions from '../../actions/AppActions'
+import AppStore from '../../stores/AppStore'
+
+export default class DisplayMessage extends Component {
+        constructor(props){
+        super(props);
+        this.state ={
+           displayArchives: AppStore.getOpenArchive(),
+           personalMessage: AppStore.getPersonalMessage(),
+           seenUsers: AppStore.getSeenUsers()     
+        };
+         this._onChange= this._onChange.bind(this)
+    }
+
+    componentWillMount ()  {
+     AppStore.addChangeListener(this._onChange);
+    }
+    componentWillUnmount () {
+      AppStore.removeChangeListener(this._onChange);
+    } 
+
+  render() {
+        if(this.state.displayArchives == 'archive') { 
+            var display = <ArchivedBox />     
+        } else if (this.state.displayArchives == 'inbox'){      
+           var display =   this.state.personalMessage.map((message, index) => {
+                        return(             
+                          <ReadMessage message={message} key={index} seenUsers={this.state.seenUsers}/>      
+                        ) 
+                    })            
+        } else if(this.state.displayArchives == 'group'){
+            var display = <MessageBoard />  
+        }
+
```

Comment:
You should do this in a function outside `render`.
Also the function can just return a result after evaluating the condition rather than storing in a variable.